lark/tree.py

```diff
@@ -211,7 +210,21 @@ def scan_values(self, pred: 'Callable[[Branch[_Leaf_T]], bool]') -> Iterator[_Le
             else:
                 if pred(c):
                     yield c
+                    
+    def replace_nodes(self, pred: 'Callable[[Token], Optional[Token]]') -> None:
+        """replace tokens in the tree using the result of pred(token) when it is not none.
```

Comment:
The description here `it is not none` does not match the implemented behavior `isinstance(node, Token)`. I think you should stick with `is not none` and change the implementation, sometimes there is a reason to use none-Token nodes. (Not sure what to do about the type hints).

Reply:
fixed 6194e05b6175022de664c9d2d1c3e92f21a88713